### PR TITLE
Documentation updates

### LINKED ISSUE #541: Improve Docs on Filtering for empty values
It's often needed to filter for items where a value **is not set**. 

It's a bit tricky because we need to distinguish between a `null` and a merely empty value. 

There are various approaches (add an extra filter, add a `MAGIC` value, ...). None are complicated, but none are such that we want to add every possible `Filter` subclass to the core library. 

(If a decent generic solution turns up, we'll have that. Until then...)  

* [ ] Better document existing approaches with examples of creating your own subclass. 
    * Using `isnull`
    * None/Any handling on `ChoiceFilter` #519
    * Suggested `EmptyStringFilter` #478 
    * Suggested `NullableDateRangeFilter` #446

### PR DESCRIPTION
**Pull request by rpkilby**
Trying to reorganize some of the docs. This is ongoing, preview docs [here](http://rpkilby.github.io/django-filter/).

**TODO**:
- [x] Resolve #541 

**Notable changes**:
- Separate guide/reference/developer docs into separate sections. The ToC/sidebar should be more organized.
- Normalized some of the chapter/section names
- Separate out the tips from the main usage docs

**Minor Changes**:
- Extended toctree depth to 2 for guide
- Updated DRF supported version
- Updated/modernized testing docs
- Prepend `$` to shell commands
- Remove deprecated docs

### CODE REVIEW

**Comment by carltongibson** [comment]
So the solution to #541 could go in _Tips_...

**Comment by carltongibson** [comment]
Ah, yes... http://rpkilby.github.io/django-filter/guide/tips.html#filtering-by-empty-values 😀

**Comment by carltongibson** [comment]
Fine. Let me know when you've had enough... I'm basically happy to merge this as is — it's a big improvement — so as soon as you loose impetus let's we'll have it.

**Comment by rpkilby** [comment]
Yep - just tacking on #541, then I'm done.

**Comment by rpkilby** [comment]
Hey. I think I'm done with the docs, barring suggestions. 

I'm not too keen on the nullable date filter. It's a bit long and unwieldy, as well as not very polished. My opinion is that users should just add an extra filter that handles the `isnull` lookup instead of a complicated `MultiWidget` solution.

**Comment by carltongibson** [comment]
Fine. Say that then. (Maybe a link... – but up to you.)

**Comment by rpkilby** [comment]
That works too - adding.

**Comment by rpkilby** [comment]
Hey @carltongibson - added it into the mix.

File: docs/guide/tips.txt [hunk 1]
```diff
@@ -88,3 +88,128 @@ uncategorized products and products for a set of categories:
...
+.. code-block:: python
+
+    class ProductFilter(django_filters.FilterSet):
+        category = django_filters.ModelChoiceFilter(
+            name='category', lookup_expr='isnull',
+            null_label='Uncategorized',
+            queryset=Category.objects.all(),
+        )
+
+Solution 3: Comining fields w/ ``MultiValueField``
```
Comment (carltongibson) [inline docs/guide/tips.txt]: Typo?

**Comment by carltongibson** [comment]
OK. Awesome. Great work.

It looks like 1.0 tomorrow then. 🙂

**Pull request merged**